internal/db/bundb/bundb.go

```
@@ -123,16 +125,51 @@ func NewBunDBService(ctx context.Context, c *config.Config) (db.DB, error) {
 		conn = WrapDBConn(bun.NewDB(sqldb, pgdialect.New()))
 	case dbTypeSqlite:
 		// SQLITE
+		var err error
 
 		// Drop anything fancy from DB address
 		c.DBConfig.Address = strings.Split(c.DBConfig.Address, "?")[0]
 		c.DBConfig.Address = strings.TrimPrefix(c.DBConfig.Address, "file:")
 
+		// if we are about to try to create/open a sqlite database file, we should do some checking to ensure
```

IMO you can put all this logic in the `util` package and just have it return an error if there's a problem, instead of printing warnings. Then the caller can choose to print the error as a warning and continue, or just return an error and fail. We'll probably need this kind of checking logic elsewhere so it makes sense to make it reusable.

---

:+1: